domain/schema/secret.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package schema
+
+import (
+	"github.com/juju/juju/core/database/schema"
+)
+
+func secretSchema() schema.Patch {
+	return schema.MakePatch(`
+-- TODO: we need to ensure the label is unique for the application and all its units!!!
+-- The unique index ensures that the label is unique for the application.
+-- TRIGGER might be able to ensure the label is unique for all the units, but it's not easy.
+-- Because we need to provide all its unit UUIDs in the trigger and provide the application_uuid in the "secret_unit_owned" trigger.
+-- Therefore, we might have to do this check in service code.
```

Comment:
Or as @tlm suggested, we can have a `secret_label_ref_count` table.
The table has two columns: `applicaiton_uuid` and `label` with a unique index on these two columns.
So whenever we have a label set to either a unit-owned or application-owned secret, we will create one record in the table.

Reply:
We will ensure label uniqueness at the code level.